internal/cluster/discovery_test.go

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) 2008-2023, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cluster_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	pubcluster "github.com/hazelcast/hazelcast-go-client/cluster"
+	"github.com/hazelcast/hazelcast-go-client/cluster/discovery"
+	"github.com/hazelcast/hazelcast-go-client/internal/cluster"
+	"github.com/hazelcast/hazelcast-go-client/internal/logger"
+)
+
+type dummyDiscoverStrategy struct {
+	nodes []discovery.Node
+}
+
+func (d dummyDiscoverStrategy) DiscoverNodes(ctx context.Context) ([]discovery.Node, error) {
+	return d.nodes, nil
+}
+
+func TestDiscoveryStrategyAdapter_Addresses(t *testing.T) {
```

Comment:
Tests look ok but won't we port any java test?

Reply:
The tests in `discovery_test.go` and `discovery_it_test.go` seem to cover the important ones. Discovery strategy in the Go client is more limiited, like the metadata doesn't exist.